lib/model.js

```diff
@@ -1612,6 +1613,15 @@ module.exports = (function() {
 
     options.type = QueryTypes.BULKUPDATE;
 
+    // Remove values that are not in the options.fields
+    if (options.fields && options.fields instanceof Array) {
+      Object.keys(values).forEach(function(key) {
+        if (options.fields.indexOf(key) < 0) {
+          delete values[key];
```

Comment:
@BridgeAR 
Nope, ``values[key]=undefined`` will not fly. Delete and assignment to ``undefined`` give different results.

``
var o = {a: 1, b: 2};
o['a'] = undefined;
o = {a: undefined, b: 2}
delete o.a;
o = {b:2}
``
And, when ``undefined`` passes through, it becomes ``null`` in DB.